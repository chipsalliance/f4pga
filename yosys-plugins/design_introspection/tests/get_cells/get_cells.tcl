yosys -import
if { [info procs get_cells] == {} } { plugin -i design_introspection }
yosys -import  ;# ingest plugin commands

read_verilog $::env(DESIGN_TOP).v
# Some of F4PGA expects eblifs with only one module.
synth_xilinx -flatten -abc9 -nosrl -noclkbuf -nodsp


set fp [open [test_output_path "get_cells.txt"] "w"]

puts "\n*inter* cells quiet"
puts $fp "\n*inter* cells quiet"
puts $fp [get_cells -quiet *inter*]

puts "\n*inter* cells"
puts $fp "\n*inter* cells"
puts $fp [get_cells *inter*]

puts "\n*inter* cells with invalid filter expression"
puts $fp "\n*inter* cells with invalid filter expression"
puts $fp [get_cells -filter {mr_ff != true} *inter* ]

puts "\nFiltered cells"
puts $fp "\nFiltered cells"
puts $fp [get_cells -filter {mr_ff == true || async_reg == true && dont_touch == true} ]

puts "\nAll cells"
puts $fp "\nAll cells"
puts $fp [get_cells]

close $fp
