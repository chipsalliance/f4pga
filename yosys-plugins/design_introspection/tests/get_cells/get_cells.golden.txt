
*inter* cells quiet
bottom_intermediate_inst.OBUF_8

*inter* cells
bottom_intermediate_inst.OBUF_8

*inter* cells with invalid filter expression
bottom_intermediate_inst.OBUF_8

Filtered cells
OBUFTDS_2

All cells
{$abc$1984$lut$not$aiger1983$1} {$auto$alumacc.cc:485:replace_alu$1385.slice[0].carry4} {$auto$alumacc.cc:485:replace_alu$1385.slice[1].carry4} {$auto$alumacc.cc:485:replace_alu$1385.slice[2].carry4} {$auto$alumacc.cc:485:replace_alu$1385.slice[3].carry4} {$auto$alumacc.cc:485:replace_alu$1385.slice[4].carry4} {$auto$alumacc.cc:485:replace_alu$1385.slice[5].carry4} {$auto$alumacc.cc:485:replace_alu$1385.slice[6].carry4} {$auto$simplemap.cc:420:simplemap_dff$1471} {$auto$simplemap.cc:420:simplemap_dff$1472} {$auto$simplemap.cc:420:simplemap_dff$1473} {$auto$simplemap.cc:420:simplemap_dff$1474} {$auto$simplemap.cc:420:simplemap_dff$1475} {$auto$simplemap.cc:420:simplemap_dff$1476} {$auto$simplemap.cc:420:simplemap_dff$1477} {$auto$simplemap.cc:420:simplemap_dff$1478} {$auto$simplemap.cc:420:simplemap_dff$1479} {$auto$simplemap.cc:420:simplemap_dff$1480} {$auto$simplemap.cc:420:simplemap_dff$1481} {$auto$simplemap.cc:420:simplemap_dff$1482} {$auto$simplemap.cc:420:simplemap_dff$1483} {$auto$simplemap.cc:420:simplemap_dff$1484} {$auto$simplemap.cc:420:simplemap_dff$1485} {$auto$simplemap.cc:420:simplemap_dff$1486} {$auto$simplemap.cc:420:simplemap_dff$1487} {$auto$simplemap.cc:420:simplemap_dff$1488} {$auto$simplemap.cc:420:simplemap_dff$1489} {$auto$simplemap.cc:420:simplemap_dff$1490} {$auto$simplemap.cc:420:simplemap_dff$1491} {$auto$simplemap.cc:420:simplemap_dff$1492} {$auto$simplemap.cc:420:simplemap_dff$1493} {$auto$simplemap.cc:420:simplemap_dff$1494} {$auto$simplemap.cc:420:simplemap_dff$1495} {$auto$simplemap.cc:420:simplemap_dff$1496} {$iopadmap$top.clk} OBUFTDS_2 OBUF_6 OBUF_7 OBUF_OUT bottom_inst.OBUF_10 bottom_inst.OBUF_11 bottom_inst.OBUF_9 bottom_intermediate_inst.OBUF_8
